tests/js/lib/errors.js

```diff
@@ -0,0 +1,17 @@
+// Simple function to examine validation errors.
+"use strict";
+var fluid = fluid || require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+var jqUnit = require("jqUnit");
+
+require("../../../src/js/common/validate");
+
+fluid.registerNamespace("gpii.schema.tests");
+gpii.schema.tests.hasFieldErrors = function (results, fields) {
+    fluid.each(fields, function (field) {
+        var path = typeof field === "string" ? [field] : field;
+        var target = gpii.schema.validator.resolveTargetFromPath(results.fieldErrors, path);
```

Comment:
I think the behaviour of resolveTargetFromPath in terms of modifying its first argument might in the end prove surprising to unwary callers. Perhaps a shallow clone might be better, or at least a prominent docs warning.